File: backend/src/main/java/br/ufpr/sept/so2/modules/solicitacoes/api/dto/CriarSolicitacaoRequest.java
package br.ufpr.sept.so2.modules.solicitacoes.api.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.util.Map;

public record CriarSolicitacaoRequest(
        @NotBlank String tipoCodigo,
        @NotNull Map<String, Object> payload
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/solicitacoes/api/dto/CriarSolicitacaoRequest.kt
package br.ufpr.sept.so2.modules.solicitacoes.api.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class CriarSolicitacaoRequest(
    @field:NotBlank
    val tipoCodigo: String,
    @field:NotNull
    val payload: Map<String, Any?>,
)
